physical/consul/consul_service_discovery.go

```diff
@@ -0,0 +1,597 @@
+package consul
```

Comment:
Could this file (and its test) live in `vault/servicediscovery/consul/` instead?

Reply:
This is a good idea, but what happened when I tried to implement it is that we end up need to make almost all of the various member fields in `ConsulServiceRegistration` public, because they are referenced inside of the test suite in `physical/consul'.  I'm not sure if we want to do that, since it exposes a lot of fields that were once private.  Thoughts?

Reply:
Could we just move the service registration related tests to the new package too?

Reply:
done!